src/MarlinSimulator/hardware/Gpio.h

```diff
@@ -159,5 +166,27 @@ class Gpio {
     return pin_map[pin].attach(args...);
   }
 
+  static void resetLogs() {
```

Comment:
This function is not thread-safe. I considered adding a mutex to protect the data structure, but that would require mutex acquisition by every pin access.

I thought you might already have something I missed to allow synchronizing with the entire kernel execution system. Ideally we could perform the resetting atomically between execution steps, without requiring constant mutex acquisition.